UI : fix design discrepancies

alexis-ld opened the pull request:
**Component**:
ui, nodes, volumes

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**:  We want to align the UI better to the design following specs in #3121 .

**Summary**:

* Fixed table tooltips top row display issue. 
* Small adjustments all over the nodes/volumes pages (cursor changes, icons, ...)
* Reworked the create node and create volume pages so they fit the design better (global alignment, smaller checkboxes, toggle instead of checkbox, display required fields, ...) (#3121 requested wider inputs in node creation page, it is done, however this has not been replicated on volume creation page as the layout is a bit different and sometime has two input next to each other..)
![Screenshot 2021-03-10 at 19 02 46](https://user-images.githubusercontent.com/3480526/110811324-e2f70480-8286-11eb-97e6-c213eb9f397c.png)

* Following a discussions with @ChengYanJin I split the the VolumeListTable component in two, the layout / behavior was getting two different between the Volume page and the Node page use of this component. The new created component (NodePageVolumesTable) is much lighter and readable because I removed the need for conditional Search and Node column display. It also allowed to fix some display issue related to table size more easily.
* Added a loader next to the Node avg metrics toggle


As suggested in #3121  points 17 & 18 will come in another PR.

Screenshots:
NodeVolumeTab:
<img width="1435" alt="NodeVolumeTable" src="https://user-images.githubusercontent.com/18453133/110852075-2d41ab00-82b2-11eb-8e79-8c85bf6cadf6.png">

CreateVolume:
<img width="1438" alt="VolumeCreation" src="https://user-images.githubusercontent.com/18453133/110852080-2e72d800-82b2-11eb-99c6-df13f1f2425b.png">

VolumePage:
<img width="1438" alt="Volumelist" src="https://user-images.githubusercontent.com/18453133/110852083-2fa40500-82b2-11eb-8148-8e46b89647dd.png">

---

<!-- Declare one or more issues to close once this PR gets merged -->



Closes: #3121

alexis-ld committed:
ui,volumes: Display volume name in blue

Refs: #3121

alexis-ld committed:
ui,volumes: Make only volume name clickable instead of all row

Refs: #3121

alexis-ld committed:
ui,volumes: Fix tooltip display for first row

First row tooltip was displayed outside of the table container and was not visible.
We now display the first row tooltip below

Refs: #3121

alexis-ld committed:
ui,nodes,pods: Add Tooltip on Advanced monitoring button link

Refs: #3121

alexis-ld committed:
ui,volumes: add action cursor on Volumes table headers

Refs: #3121

alexis-ld committed:
ui,volumes,nodes: Change create node/volume icon (remove circle around plus)

Refs: #3121

alexis-ld committed:
ui,volumes,nodes: Replace table search by core-ui's SearchInput

Adds magnifier icon as well as consistent styling

Refs: #3121

alexis-ld committed:
ui,nodes: Fix node row size changing on hover

Refs: #3121

alexis-ld committed:
ui,form: Add temporary core-ui styling

Smaller checkbox and toggle. This should eventually be moved as a size prop in core-ui

Refs: #3121

alexis-ld committed:
ui,node: Use common toggle styling component in Node page metric tab

Refs: #3121

alexis-ld committed:
ui,nodes,volumes: Align create node and volume to fit design

Small adjustements, wider inputs, smaller checkboxes, required fields, ...

Refs: #3121

alexis-ld committed:
ui,nodes: Display a Loader while loading node avg metrics

Refs: #3121

alexis-ld committed:
ui,volumes: Fix first row tooltip for name and node fields

Was getting under table header and was not visible

Refs: #3121

alexis-ld committed:
ui,volumes: Split volume table in two component

The same component was used for volume page list and node page volume tab.

Split into two components for more flexibility and more readable code

Refs: #3121

bert-e commented:
# Hello alexis-ld,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/3121-fit-to-design` into
`development/2.8`
* [w/2.9/improvement/3121-fit-to-design](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/improvement/3121-fit-to-design) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

Cuervino commented:
1. Nodes page, pods tab > the logs column is not aligned.
Moreover, the header of "Status" and "Namespace" columns are not aligned (left) with the content.
![image](https://user-images.githubusercontent.com/28654802/111176532-b524ff00-85a9-11eb-9e57-c3cf925ad6da.png)

2. In the forms, the sentence "All * are mandatory fields" is in bold, it should be in regular font.

3. For the form "create new node", I still see issue 12 (The creation button is not visible), and there is a ghost slider (on Firefox).
![image](https://user-images.githubusercontent.com/28654802/111176558-bb1ae000-85a9-11eb-9e92-c897e60cad83.png)

4. Node page, volumes tab > the name are in blue, but should not be underlined (for instance, it's not the case in the Volumes page, overview tab)
5. Table on Volumes tab > too much aligned on the right (the sorting icon that is displayed on hover is outside of the table.
I would leave at least 8px before the end of the area.
![image](https://user-images.githubusercontent.com/28654802/111176707-dc7bcc00-85a9-11eb-81e8-307ac08a1b05.png)

6. Remark: The tooltip on hovering when content overflow could be displayed on a single line (especially that the content here is not very long).
The tooltip is displayed even when there is no truncation, I don't see the added value.
![image](https://user-images.githubusercontent.com/28654802/111176598-c3731b00-85a9-11eb-9673-8954dc0b00ed.png)

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/3121-fit-to-design` has diverged from
`development/2.8` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.8` into `improvement/3121-fit-to-design`
* Rebase `improvement/3121-fit-to-design` onto `origin/development/2.8`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

ChengYanJin commented on ui/src/components/TableRow.js:
```diff
@@ -36,12 +35,11 @@ export const UnknownIcon = styled.i`
   return (
     <TableRowStyle
       {...row.getRowProps({
-        onClick: () => onClickRow(row),
```
Would be cool to remove the highlight of the row, since we won't be able to select the row anymore. 
@Cuervino Can you confirm that please?

![image](https://user-images.githubusercontent.com/18453133/111185317-223c9280-85b2-11eb-8943-61c4051e1373.png)

Cuervino replied:
Yes, indeed, here selecting the line doesn't have any added value. Moreover, the table in the "Pods" tab already has this "non-selectable lines" logic.

ChengYanJin commented on ui/src/components/VolumeListTable.js:
```diff
@@ -96,6 +96,10 @@ const ActionContainer = styled.span`
 `;
 
+const NameLinkContaner = styled.div`
+  cursor: pointer;
```
1. We should remove the underline of the volume name.
2. We should add a margin between the volume name and usage progressbar.

ChengYanJin commented on ui/src/components/NodePagePodsTab.js:
```diff
@@ -184,13 +188,18 @@ const NodePagePodsTab = (props) => {
-              rel="noopener noreferrer"
+            <Tooltip
+              placement={'left'}
+              overlay={<TooltipContent>Advanced Monitoring</TooltipContent>}
```
would be great to add the translations

ChengYanJin commented on ui/src/components/VolumeListTable.js:
```diff
@@ -126,28 +132,14 @@ function GlobalFilter({
 
   return (
-    <input
+    <SearchInput
```
the font family of the search input should be `Lato`

ChengYanJin commented on ui/src/components/style/FormStyle.js:
```diff
@@ -0,0 +1,44 @@
+  }
+`;
+
+export const ToggleWrapper = styled.div`
```
Would make sense to create a PR in core-ui for the `CheckboxWrapper` and `ToggleWrapper`

alexis-ld replied:
@ChengYanJin the other components in core-ui have a "size" prop with the usual variants (smaller, small, base, large, larger) . I don't think this is relevant for the checkbox of the toggle, it is unlikely we need a very larger checkbox anywhere... ^^ do you have any preference on how to implement this then ? a "isSmall" prop or something like that ?

ChengYanJin commented on ui/src/containers/CreateVolume.js:
```diff
@@ -150,12 +157,6 @@ const LabelsName = styled(LabelsValue)`
   color: ${(props) => props.theme.brand.textPrimary};
 `;
 
-const PageContainer = styled.div`
-  display: flex;
-  // To avoid the big empty space on the right side of page
-  justify-content: space-around;
-`;
-
```
We should keep this styling to put this creation form in the middle of the page to avoid the big empty space on the right side of the page.

alexis-ld replied:
Oh ok. On the design it was aligned left : 
![Screenshot 2021-03-16 at 15 37 35](https://user-images.githubusercontent.com/3480526/111327325-95efa580-866d-11eb-812c-2610fb5578fb.png)

Will move it back to the center.

ChengYanJin replied:
@alexis-ld 
thank you! this should be the correct one!
![image](https://user-images.githubusercontent.com/18453133/111436970-dd744100-8702-11eb-91df-f1c8b2b8bf62.png)

ChengYanJin commented on ui/src/containers/CreateVolume.js:
```diff
@@ -514,19 +516,23 @@ const CreateVolume = (props) => {
               return (
                 <Form>
+                  <FormSection>
+                    <RequiredText>All * are mandatory fields</RequiredText>
```
the text might need to be slightly changed to, please add translation as well.
`* Required fields`

ChengYanJin reviewed (changes requested):
Thanks for changing the target to `development/2.9`
And address some visual changes.

In addition: 
I think we haven't addressed this issue:
NodeCreation:12. The creation button is not visible
![image](https://user-images.githubusercontent.com/18453133/111191934-d6d9b280-85b8-11eb-98aa-48872f238446.png)
We can't scroll down to see the button at this moment.

alexis-ld committed:
ui,volumes,nodes: Remove local Toggle and Checkbox styling

Moved those styles to core-ui for consistency

Refs: #3121

alexis-ld committed:
ui,volumes,nodes: Update Required text styling + translations

Refs: #3121

alexis-ld committed:
ui,volumes: Remove volume row highlight when non clickable

Refs: #3121

alexis-ld committed:
ui,volumes: Remove volume link underline and add right margin to progressbar

Refs: #3121

alexis-ld committed:
ui,pods: Use translation for advanced mon. tooltip

Refs: #3121

alexis-ld committed:
ui: Use Lato font in inputs

Refs: #3121

alexis-ld committed:
ui,volumes,nodes: Move creation forms to center of the page

Refs: #3121

alexis-ld committed:
ui,volumes,nodes: Fix creation forms vertical scroll on small screens

Bottom action buttons were not visible on small screens.

Refs: #3121

alexis-ld committed:
ui,volumes: Add ConstrainedText for node tab volumes table volume name

Refs: #3121

alexis-ld committed:
ui,volumes: Fix node tab table header carets

Health sort caret was overlapping with next column. Latency caret was overflowing container on the right side.

Refs: #3121

bert-e commented:
# History mismatch



Merge commit #9564030b363d43d2fda4d456eff20ccf80b1df73 on the integration branch
`w/2.9/improvement/3121-fit-to-design` is merging a branch which is neither the current
branch `improvement/3121-fit-to-design` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `improvement/3121-fit-to-design` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/3121-fit-to-design` into
`development/2.8`
* [w/2.9/improvement/3121-fit-to-design](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/improvement/3121-fit-to-design) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

alexis-ld commented:
@ChengYanJin @Cuervino I pushed the changes.

The table scrollbar issue seems a bit more complex than expected and I will push the fix in another PR so we can move on with this one.

alexis-ld commented:
Also the toggle/tooltip/checkbox fixes have been moved to core-ui directly. So for this to work properly we will need to merge https://github.com/scality/core-ui/pull/272 first and use this version in metal :)

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/3121-fit-to-design` has diverged from
`development/2.9` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.9` into `improvement/3121-fit-to-design`
* Rebase `improvement/3121-fit-to-design` onto `origin/development/2.9`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

ChengYanJin commented on ui/src/components/TableRow.js:
```diff
@@ -1,16 +1,20 @@
-    outline: none;
-    cursor: pointer;
-  }
+  ${(props) =>
```
I feel by doing this will create complexity when we will extract the table component.
Basically, we will extract at least 2 table components. One simple: like the one in the volume tab.  Another one with more functionalities, maybe row selection + search, etc. 

So here, I would purpose to have another <TableRow/> which be used in the simple use case. 
I know we will duplicate the code for the moment, but that's OK.

ChengYanJin reviewed (changes requested):
Thanks for taking care of the change request from me and Valamir!

I see three remain issues:
1. Would be great to leverage the empty space be to used by the Volume Name, whose field usually has longer text. 
![image](https://user-images.githubusercontent.com/18453133/111803943-d9494e80-88cf-11eb-9b2e-6a4fc2ef7775.png)
2. the loader of when active cluster average should be next to the text.
![image](https://user-images.githubusercontent.com/18453133/111804344-478e1100-88d0-11eb-8b02-a12c78131790.png)
3. the documentation tooltip is not visible.
![image](https://user-images.githubusercontent.com/18453133/111805429-4f01ea00-88d1-11eb-9250-b98041f5bbd1.png)

alexis-ld committed:
ui,layout: Add ScrollbarWrapper from core-ui

Refs: #3121

alexis-ld committed:
ui,nodes: Move node avg loader on the toggle level

Refs: #3121

alexis-ld committed:
ui,volumes: Update documentation icon tooltip location

Tooltip was overflowing and was not visible

Refs: #3121

alexis-ld committed:
ui,volumes: Duplicate table row component in duplicated volume table

To avoid using the same TableRow component to make the component extraction easier

Refs: #3121

alexis-ld committed:
ui,volumes: Make more room for the name field in Volumes table

Refs: #3121

alexis-ld committed:
ui,nodes,pods: Remove unneeded padding on Pods table headers

Refs: #3121

alexis-ld committed:
ui,volumes: Smaller usage progress bar

To give more space to the other fields on small screens

Refs: #3121

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

alexis-ld commented:
@ChengYanJin Should be all good now :) 

Also please note that since we moved the Custom Scrollbar rules to its own component in core-ui, these changes won't work before https://github.com/scality/core-ui/pull/274 is merged and this version of core-ui is used within Metal :)

alexis-ld committed:
ui: Upgrade core-ui to 0.14.0

alexis-ld commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin







*The following options are set:* **approve**

alexis-ld committed:
ui,nodes: Remove container overflow property for node detail

This makes the content use the full parent height

Refs: #3121

alexis-ld committed:
ui,volumes: Fix details tab scroll

By setting a defined height on the container making it scrollable within its parent flex element

alexis-ld committed:
ui,nodes: Fix pods tab scroll height

Refs: #3121

alexis-ld committed:
ui: upgrade core-ui to 0.14.1

ChengYanJin reviewed (approved):
LGTM!

alexis-ld commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/de1cb5f78281a65ce7108093bb4890367a413327) did not succeed in branch improvement/3121-fit-to-design.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.9`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.9`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`



Please check the status of the associated issue None.

Goodbye alexis-ld.
